tests/models/idefics2/test_processing_idefics2.py

```diff
@@ -233,3 +261,17 @@ def test_apply_chat_template(self):
             "Assistant:"
         )
         self.assertEqual(rendered, expected_rendered)
+
+    def prepare_text_inputs(self, batched=False):
+        if batched:
+            return ["<image> lower newer", "<image> upper older longer string"]
+        return "<image> lower newer"
+
+    @require_vision
+    def prepare_image_inputs(self, batched=False):
```

Comment:
Rather than `batched`, I'd have this as an optional int, which returns a single image if not set and a list of `n` images if set